sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala

```diff
@@ -78,4 +78,29 @@ class ParquetEncodingSuite extends ParquetCompatibilityTest with SharedSQLContex
       }}
     }
   }
+
+  test("Read row group containing both dictionary and plain encoded pages") {
+    spark.conf.set("parquet.dictionary.page.size", "2048")
+    spark.conf.set("parquet.page.size", "4096")
+
+    withTempPath { dir =>
+      // In order to explicitly test for SPARK-14217, we set the parquet dictionary and page size
+      // such that the following data spans across 3 pages (within a single row group) where the
+      // first page is dictionary encoded and the remaining two are plain encoded.
+      val data = (0 until 512).flatMap(i => List(i.toString, i.toString, i.toString))
+      data.toDF("f").coalesce(1).write.parquet(dir.getCanonicalPath)
+      val file = SpecificParquetRecordReaderBase.listDirectory(dir).toArray.head
+
+      val reader = new VectorizedParquetRecordReader
+      reader.initialize(file.asInstanceOf[String], null)
```

Comment:
This is calling into java code, so named parameters wouldn't work. I added a comment to make it clear.